test/lib/ui5framework/npm/Installer.js

```diff
@@ -0,0 +1,179 @@
+});
+
+test.serial("Installer: _getLockPath", async (t) => {
+	const installer = new Installer({
+		cwd: "/cwd/",
+		ui5HomeDir: "/ui5Home/"
+	});
+
+	const lockPath = installer._getLockPath({
+		pkgName: "@openui5/sap.ui.lib1",
+		version: "1.2.3"
+	});
+
+	t.is(lockPath, path.join("/ui5Home/", "framework", "locks", "package-@openui5-sap.ui.lib1@1.2.3.lock"));
+});
+
+test.serial("Installer: _synchronize", async (t) => {
+	const installer = new Installer({
+		cwd: "/cwd/",
+		ui5HomeDir: "/ui5Home/"
+	});
+
+	t.context.mkdirpStub.resolves();
+	t.context.lockStub.resolves();
+	t.context.unlockStub.resolves();
+
+	const getLockPathStub = sinon.stub(installer, "_getLockPath").returns("/locks/lockfile.lock");
+
+	const callback = sinon.stub().resolves();
+
+	await installer._synchronize({
+		pkgName: "@openui5/sap.ui.lib1",
+		version: "1.2.3"
+	}, callback);
+
+	t.is(getLockPathStub.callCount, 1, "_getLockPath should be called once");
+	t.deepEqual(getLockPathStub.getCall(0).args, [{pkgName: "@openui5/sap.ui.lib1", version: "1.2.3"}],
+		"_getLockPath should be called with expected args");
+
+	t.is(t.context.mkdirpStub.callCount, 1, "_mkdirp should be called once");
+	t.deepEqual(t.context.mkdirpStub.getCall(0).args, [path.join("/ui5Home/", "framework", "locks")],
+		"_mkdirp should be called with expected args");
+
+	t.is(t.context.lockStub.callCount, 1, "_lock should be called once");
+	t.deepEqual(t.context.lockStub.getCall(0).args, [
+		"/locks/lockfile.lock", {wait: 10000, stale: 60000, retries: 10}
+	], "_lock should be called with expected args");
+
+	t.is(t.context.unlockStub.callCount, 1, "_unlock should be called once");
+	t.deepEqual(t.context.unlockStub.getCall(0).args, ["/locks/lockfile.lock"],
+		"_unlock should be called with expected args");
+
+	t.is(callback.callCount, 1, "callback should be called once");
+
+	t.true(t.context.lockStub.calledBefore(callback), "Lock should be called before invoking the callback");
+	t.true(t.context.unlockStub.calledAfter(callback), "Unlock should be called after invoking the callback");
+});
+
+test.serial("Installer: _synchronize should unlock when callback promise has resolved", async (t) => {
+	t.plan(4);
+
+	const installer = new Installer({
+		cwd: "/cwd/",
+		ui5HomeDir: "/ui5Home/"
+	});
+
+	t.context.mkdirpStub.resolves();
+	t.context.lockStub.resolves();
+	t.context.unlockStub.resolves();
+
+	sinon.stub(installer, "_getLockPath").returns("/locks/lockfile.lock");
+
+	const callback = sinon.stub().callsFake(() => {
+		t.is(t.context.lockStub.callCount, 1, "_lock should have been called when the callback is invoked");
+		return Promise.resolve().then(() => {
+			t.is(t.context.unlockStub.callCount, 0,
+				"_unlock should not be called when the callback did not fully resolve, yet");
+		});
+	});
+
+	await installer._synchronize({
+		pkgName: "@openui5/sap.ui.lib1",
+		version: "1.2.3"
+	}, callback);
+
+	// Ensure to wait for the callback promise to be resolved
```

Comment:
Why? Can _synchronize resolve *before* the callback has resolved?

Reply:
Yes, if it doesn't use `await` on it. But waiting for the promise still doesn't really make sense